main.cpp

```diff
@@ -38,17 +41,16 @@ void find_k_to_end(struct Node* head, int k, int &result_value) {
         length++;
     }
     
-    if (length < k) {
-        return;
+    if (length < k - 1) {
```

Comment:
It's actually 'length < k + 1', which I think can be more clearly expressed as 'k > length - 1' or ' k >= length'